gui/src/app/FileEditor/TextEditor.tsx

```diff
@@ -27,6 +27,8 @@ type Props = {
   toolbarItems?: ToolbarItem[];
   label: string;
   codeMarkers?: CodeMarker[];
+  hintTextOnEmpty?: string;
```

Comment:
Would it be possible to have this be something like `string | HTMLSpanElement`?

In particular, I think it would be nice to have multiple links or be able to do what VSCode does an have some of the text be a link and some be click-through, e.g

![image](https://github.com/user-attachments/assets/e16a73a3-dc24-4aea-b14d-d61eb6636980)

Only the blue portion here is clickable.

This would let us make the text something like "Define a dictionary called `data` to update the data.json. [Click here]() to generate an example"

Reply:
I like that idea, I'll work on it, and make sure that the styling etc is taken care of inside the editor component.

Reply:
@WardBrian This is implemented in my latest commit.